src/api/java/mekanism/api/math/FloatingLong.java

```diff
@@ -334,6 +338,117 @@ public FloatingLong divideEquals(FloatingLong toDivide) {
         return setAndClampValues(value, decimal);
     }
 
+    /**
+     * Divides this {@link FloatingLong} by the given unsigned long primitive, modifying the current object unless it is a constant in which case it instead returns the
+     * result in a new object. Rounds to the nearest 0.0001
+     *
+     * @param toDivide The value to divide by represented as an unsigned long.
+     *
+     * @return The {@link FloatingLong} representing the value of dividing this {@link FloatingLong} by the given {@link FloatingLong}.
+     *
+     * @throws ArithmeticException if {@code toDivide} is zero.
+     * @apiNote It is recommended to set this to itself to reduce the chance of accidental calls if calling this on a constant {@link FloatingLong}
+     * <br>
+     * {@code value = value.divideEquals(toDivide)}
+     */
+    public FloatingLong divideEquals(long toDivide) {
+        if (toDivide == 0) {
+            throw new ArithmeticException("Division by zero");
+        }
+        if (this.isZero())
+            return this;
+
+        long val = Long.divideUnsigned(this.value, toDivide);
+        long rem = Long.remainderUnsigned(this.value, toDivide);
+
+        //just need to figure out remainder -> decimal
+        long dec;
+
+        //okay, now what if rem * SINGLE_UNIT * 10L will overflow?
+        if (Long.compareUnsigned(rem * 10, MAX_LONG_SHIFT) >= 0) {
+            //if that'll overflow, then todivide also has to be big. let's just lose some denominator precision and use that
+            dec = Long.divideUnsigned(rem, Long.divideUnsigned(toDivide, SINGLE_UNIT * 10L)); //same as multiplying numerator
```

Comment:
This crashes with divide by zero when `Long.divideUnsigned(toDivide, SINGLE_UNIT * 10L)` evaluates to zero which happens when `toDivide < SINGLE_UNIT * 10L`.
Example of an input that causes this to happen is.
```java
FloatingLong.create(0, (short) 1).divide(2L);
```

I think one of the reasons why this is happening is because of:
```java
private static final long MAX_LONG_SHIFT = -1L / SINGLE_UNIT;
```
Evaluating is to zero because it's operating on a signed long.
Which means the condition:
```java
if (Long.compareUnsigned(rem * 10, MAX_LONG_SHIFT) >= 0)
```
Is often evaluating to true (any remainder that is non-zero).
Didn't seem to get caught by the Property tests since the randomly generated samples might have not have run into the situation?

Reply:
Thanks for catching the MAX_LONG_SHIFT error, I definitely will fix that. But I do not think there is any case of divide_by_zero in the `if (Long.compareUnsigned(rem * 10, MAX_LONG_SHIFT) >= 0)` case you mentioned. 

This is because the above if statement guarantees that the remainder (rem) is very large (larger than MAX_LONG_SHIFT / 10). Therefore, toDivide must also be very large (> rem+1). In this case, it cannot go to zero with the following
`Long.divideUnsigned(toDivide, SINGLE_UNIT * 10L)`

If you think it's hard to read through the logic, I'm all for changing it to add readability. Just let me know if you think it's required and I have no issues :)

Regarding the discrepancy, I will add a fix for the discrepancies, I agree they should be equal.

Reply:
Yes, sorry, it doesn't crash with that change to using unsigned division added. I should have made it clearer by seperating out the second issue (which I ran into after switching to unsigned division in testing but it was not related).

Reply:
Resolved with https://github.com/mekanism/Mekanism/pull/6988/commits/e49f3bf423269668e3dda01a49cf51ac281ce31e